text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,329 @@
+
+```rust, ignore
+decl_storage! {
+	trait Store for Module<T: Trait> as Example {
+		// --snip--
+
+		/// Permill accumulator, value starts at 1 (multiplicative identity)
+		PermillAccumulator get(fn permill_value): Permill = Permill::one();
+	}
+}
+```
+
+The only extrinsic for this Permill accumulator is the one that allows users to submit new `Permill` values to get multiplied into the accumulator.
+
+```rust, ignore
+fn update_permill(origin, new_factor: Permill) -> DispatchResult {
+	ensure_signed(origin)?;
+
+	let old_accumulated = Self::permill_value();
+
+	// There is no need to check for overflow here. Permill holds values in the range
+	// [0, 1] so it is impossible to ever overflow.
+	let new_product = old_accumulated.saturating_mul(new_factor);
+
+	// Write the new value to storage
+	PermillAccumulator::put(new_product);
+
+	// Emit event
+	Self::deposit_event(Event::PermillUpdated(new_factor, new_product));
+	Ok(())
+}
+```
+
+The code of this extrinsic largely speaks for itself. One thing to take particular note of is that we _don't_ check for overflow on the multiplication. If you've read many of the recipes you know that a Substrate runtime must never panic, and a developer must be extremely diligent in always checking for and gracefully handling error conditions. Because `Permill` only holds values between 0 and 1, we know that their product will always be in that same range. Thus it is impossible to overflow or saturate. So we can happily use `saturating_mul` and move on.
+
+### Substrate-fixed Accumulator
+
+[Substrate-fixed](https://github.com/encointer/substrate-fixed/) takes a more traditional approach in that their types represent numbers with both whole _and_ fractional parts. For this implementation, we'll use the `U16F16` type. This type contains an unsigned number (indicated by the `U` at the beginning) and has 32 _total_ bits of precision - 16 for the integer part, and 16 for the fractional part. There are several other types provided that follow the same naming convention. Some examples include `U32F32` and `I32F32` where the `I` indicates a signed number, just like in Rust primitive types.
+
+As in the `Permill` example, we begin by declaring the storage item. With substrate-fixed, there is not a `one` function, but there is a `from_num` function that we use to set the storage item's default value. This `from_num` method and it's counterpart `to-num` are your primary ways of converting between substrate-fixed types and Rust primitive types. If your use case does a lot of fixed-point arithmetic, like ours does, it is advisable to keep your data in substrate-fixed types.
+
+> We're able to use `U16F16` as a storage item type because it, and the other substrate-fixed types, implement the parity scale codec.
```

Comment:
```suggestion
> We're able to use `U16F16` as a storage item type because it, and the other substrate-fixed types, implements the parity scale codec.
```